python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py

```diff
@@ -779,6 +779,137 @@ def evaluate_with_onnx(self, data,
+                                              batch_size=batch_size)
+            return yhat
+
+        # step1, according to validation dataset, calculate inherent noise
+        # which should be done during fit
+        if not hasattr(self, "data_noise"):
+            invalidInputError(val_data is not None,
+                              "When call predict_interval for the first time, you must pass in "
+                              "validation data to calculate data noise.")
+            # data transform
+            if isinstance(val_data, TSDataset):
+                _rolled = val_data.numpy_x is None
+                val_data = val_data.to_torch_data_loader(batch_size=batch_size,
+                                                         roll=_rolled,
+                                                         lookback=self.data_config['past_seq_len'],
+                                                         horizon=self.data_config['future_seq_len'],
+                                                         feature_col=data.roll_feature,
+                                                         target_col=data.roll_target,
+                                                         shuffle=False)
+
+            if isinstance(val_data, DataLoader):
+                input_data = val_data
+                target = np.concatenate(tuple(val[1] for val in val_data), axis=0)
+            else:
+                input_data, target = val_data
+            self.internal.eval()
+            val_yhat = calculate(input_data, self.internal)
+            self.data_noise = Evaluator.evaluate(["mse"], target,
+                                                 val_yhat, aggregate=None)[0]  # 2d array
+
+        # step2: data preprocess
+        if isinstance(data, TSDataset):
+            _rolled = data.numpy_x is None
+            data = data.to_torch_data_loader(batch_size=batch_size,
+                                             roll=_rolled,
+                                             lookback=self.data_config['past_seq_len'],
+                                             horizon=self.data_config['future_seq_len'],
+                                             feature_col=data.roll_feature,
+                                             target_col=data.roll_target,
+                                             shuffle=False)
+
+        # step3: calculate model uncertainty based MC Dropout
+        def apply_dropout(m):
+            if type(m) == torch.nn.Dropout:
+                m.train()
+
+        # turn on dropout
+        self.internal.apply(apply_dropout)
+
+        y_hat_list = []
+        for i in range(repetition_times):
+            y_hat_list.append(calculate(data, self.internal))
+        y_hat_mean = np.mean(np.stack(y_hat_list, axis=0), axis=0)
+        invalidInputError(y_hat_mean.shape == y_hat_list[0].shape,
+                          "dismatch shape between y_hat_mean and y_hat")
```

Comment:
why we need this invalidinputerror check?

Reply:
will delete this check.